modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperienceLocalServiceImpl.java

```diff
@@ -236,6 +240,50 @@ public void deleteSegmentsExperiences(
 		}
 	}
 
+	@Override
+	public SegmentsExperience duplicateSegmentsExperience(
+			Locale locale, long segmentsExperienceId,
+			ServiceContext serviceContext)
+		throws PortalException {
+
+		SegmentsExperience segmentsExperience =
+			segmentsExperiencePersistence.findByPrimaryKey(
+				segmentsExperienceId);
+
+		List<SegmentsExperience> segmentsExperiences = new ArrayList<>(
+			segmentsExperiencePersistence.findByG_C_C_GEP(
+				segmentsExperience.getGroupId(),
+				segmentsExperience.getClassNameId(),
+				segmentsExperience.getClassPK(),
+				segmentsExperience.getPriority()));
+
+		for (SegmentsExperience curSegmentsExperience : segmentsExperiences) {
+			curSegmentsExperience.setPriority(
+				curSegmentsExperience.getPriority() + 1);
+
+			segmentsExperienceLocalService.updateSegmentsExperience(
+				curSegmentsExperience);
+		}
+
+		StringBundler sb = new StringBundler(5);
```

Comment:
Better if the name of the new experience come from outside the Service layer.

Reply:
wow, I like it, yes, I totally agree